Library/Homebrew/cmd/--prefix.rb

```diff
@@ -25,22 +25,44 @@ def __prefix_args
       EOS
       switch "--unbrewed",
              description: "List files in Homebrew's prefix not installed by Homebrew."
+      switch "--installed",
+             description: "Outputs nothing and returns a failing status code if <formula> is not installed."
+      conflicts "--unbrewed", "--installed"
     end
   end
 
   def __prefix
     args = __prefix_args.parse
 
+    raise UsageError, "`--installed` requires a formula argument." if args.installed? && args.no_named?
+
     if args.unbrewed?
       raise UsageError, "`--unbrewed` does not take a formula argument." unless args.no_named?
 
       list_unbrewed
     elsif args.no_named?
       puts HOMEBREW_PREFIX
     else
-      puts args.named.to_resolved_formulae.map { |f|
-        f.opt_prefix.exist? ? f.opt_prefix : f.latest_installed_prefix
+      formulae = args.named.to_resolved_formulae
+      puts formulae.map { |f|
+        if f.opt_prefix.exist?
+          f.opt_prefix
+        elsif args.installed?
+          nil
+        else
+          f.latest_installed_prefix
+        end
       }
```

Comment:
```suggestion
     prefixes = formulae.map do |f|
        if f.opt_prefix.exist?
          f.opt_prefix
        elsif args.installed?
          nil
        else
          f.latest_installed_prefix
        end
      end.compact
      puts prefixes
```

Otherwise `brew --prefix --installed $SOMETHING_NOT_INSTALLED` will always have a blank line before the error.